commands.py

```diff
@@ -177,111 +176,71 @@ def RGB(r, g, b, colour):
-                await ctx.send(embed=RGB(int(colour[1:3], 16), int(colour[3:5], 16), int(colour[5:], 16), colour))
-            elif len(colour) == 4:
-                await ctx.send(embed=RGB(int(colour[1] * 2, 16), int(colour[2] * 2, 16), int(colour[3] * 2, 16), colour))
-        elif re.fullmatch(r"([0-9A-Fa-f]{3}){1,2}", colour):
-            if len(colour) == 6:
-                await ctx.send(embed=RGB(int(colour[:2], 16), int(colour[2:4], 16), int(colour[4:], 16), colour))
-            elif len(colour) == 3:
-                await ctx.send(embed=RGB(int(colour[0] * 2, 16), int(colour[1] * 2, 16), int(colour[2] * 2, 16), colour))
-        elif re.fullmatch(r"rgb\((\d{1,3}, *){2}\d{1,3}\)", colour):
-            content = ast.literal_eval(colour[3:])
-            await ctx.send(embed=RGB(content[0], content[1], content[2], colour))
-        elif re.fullmatch(r"hsl\(\d{1,3}(\.\d*)?, *\d{1,3}(\.\d*)?%?, *\d{1,3}(\.\d*)?%?\)", colour):
-            content = colour
-            colour = colour[4:].split(")")[0].split(",")
-            h = Fraction(colour[0])
-            s = Fraction(colour[1].replace("%", ""))
-            l = Fraction(colour[2].replace("%", ""))
-
-            if h > 360 or s > 100 or l > 100:
-                return await ctx.send("Please input 0 <= h <= 360, 0 <= s, l <= 100.")
-
-            await ctx.send(embed=hslRGB(h, s, l, content))
-        elif re.fullmatch(r"cmyk\((\d{1,3}(\.\d*)?%?, *){3}\d{1,3}(\.\d*)?%?\)", colour):
-            content = colour
-            colour = colour[5:].split(")")[0].split(",")
-            c = Fraction(colour[0].replace("%", ""))
-            m = Fraction(colour[1].replace("%", ""))
-            y = Fraction(colour[2].replace("%", ""))
-            k = Fraction(colour[3].replace("%", ""))
-
-            if c > 100 or m > 100 or y > 100 or k > 100:
-                return await ctx.send("Please input 0 <= c, m, y, k <= 100.")
-
-            await ctx.send(embed=cmykRGB(c, m, y, k, content))
-        elif colour.lower() in css.keys():
+        elif c_str := re.search(r"#([0-9A-Fa-f]{3}){1,2}", colour):
+            c_str = c_str.group()
+            if len(c_str) == 7:
+                await ctx.send(embed=RGB(int(c_str[1:3], 16), int(c_str[3:5], 16), int(c_str[5:], 16), c_str))
+            elif len(c_str) == 4:
+                await ctx.send(embed=RGB(int(c_str[1] * 2, 16), int(c_str[2] * 2, 16), int(c_str[3] * 2, 16), c_str))
+        elif c_str := re.search(r"([0-9A-Fa-f]{3}){1,2}", colour):
+            c_str = c_str.group()
+            if len(c_str) == 6:
+                await ctx.send(embed=RGB(int(c_str[:2], 16), int(c_str[2:4], 16), int(c_str[4:], 16), c_str))
+            elif len(c_str) == 3:
+                await ctx.send(embed=RGB(int(c_str[0] * 2, 16), int(c_str[1] * 2, 16), int(c_str[2] * 2, 16), c_str))
+        elif c_str := re.search(r"rgb\((\d{1,3}), *(\d{1,3}), *(\d{1,3})\)", colour):
+            r, b, g = map(int, c_str.group(1, 2, 3))
```

Comment:
r, b, g?

shouldn't it be r, g, b lol

or are you obsessed with ruth bader ginsberg after pa1 🤔